crates/top/rerun/src/commands/rrd/stats.rs

```diff
@@ -0,0 +1,397 @@
+
+        println!();
+        println!("Size (data only, uncompressed)");
+        println!("------------------------------");
+        print_ipc_size_bytes_stats(ipc_data_size_bytes_uncompressed);
+
+        Ok(())
+    }
+}
+
+#[derive(Clone, Debug)]
+struct ChunkStats {
+    entity_path: String,
+
+    indexes: Vec<String>,
+    components: Vec<String>,
+
+    num_rows: u64,
+    num_indexes: u64,
+    num_components: u64,
+
+    ipc_size_bytes_compressed: u64,
+    ipc_size_bytes_uncompressed: u64,
+
+    ipc_schema_size_bytes: u64,
+    ipc_data_size_bytes: u64,
+}
+
+fn compute_stats(msg: &Msg) -> anyhow::Result<Option<ChunkStats>> {
+    if let Msg::ArrowMsg(arrow_msg) = msg {
+        let re_protos::log_msg::v1alpha1::ArrowMsg {
+            store_id: _,
+            chunk_id: _,
+            compression,
+            uncompressed_size,
+            encoding: _,
+            payload,
+        } = arrow_msg;
+
+        let mut uncompressed = Vec::new();
+
+        const COMPRESSION_NONE: i32 = re_protos::log_msg::v1alpha1::Compression::None as _;
+        const COMPRESSION_LZ4: i32 = re_protos::log_msg::v1alpha1::Compression::Lz4 as _;
+
+        let payload = match *compression {
+            COMPRESSION_NONE => payload,
+
+            COMPRESSION_LZ4 => {
+                uncompressed.resize(*uncompressed_size as _, 0);
+                re_log_encoding::external::lz4_flex::block::decompress_into(
+                    payload,
+                    &mut uncompressed,
+                )?;
+                uncompressed.as_slice()
+            }
+
+            huh => anyhow::bail!("unknown Compression: {huh}"),
+        };
+
+        let ipc_schema_size_bytes = {
+            // NOTE: This is based on the implementation of `arrow::ipc::convert::try_schema_from_ipc_buffer`.
+
+            const CONTINUATION_MARKER: [u8; 4] = [0xff; 4];
+
+            anyhow::ensure!(
+                payload.len() >= 4,
+                "The payload length is less than 4 and missing the continuation marker or length of payload"
+            );
+
+            let (len, _payload) = if payload[..4] == CONTINUATION_MARKER {
+                anyhow::ensure!(
+                    payload.len() >= 8,
+                    "The payload length is less than 8 and missing the length of payload"
+                );
+                payload[4..].split_at(4)
+            } else {
+                payload.split_at(4)
+            };
+
+            let len = <i32>::from_le_bytes(len.try_into()?);
+            anyhow::ensure!(
+                len >= 0,
+                "The encapsulated message's reported length is negative ({len})"
+            );
+
+            len as u64
+        };
+
+        let decoded = re_log_encoding::protobuf_conversions::arrow_msg_from_proto(arrow_msg)?;
+
+        let schema = decoded.batch.schema();
+
+        let entity_path = {
+            let entity_path = schema.metadata().get("rerun:entity_path");
```

Comment:
do we have access to constants for all these metadata names here? It's not like we never rename those and you dont'have any tests in here so the tool might silently break :/

Reply:
I've purposefully written this as if I was an external user building their own little independent data crunching tool. Part of the value of Sorbet is that it's supposed to give me a stable ABI that I can target: I think it's more than time to put that promise to the test, especially now that we are starting to build datasets that cannot possibly be re-regenerated from scratch.